dustinsgoodman/next-playground
# setup project dependencies for better coding environment

# Overview
This sets up code linting, testing, storybook, styled components, and other utilities to be used in this project.

# Screenshots for Mobile and Desktop views (if applicable)

## Before
![image](https://user-images.githubusercontent.com/1815379/101993884-ed0b6600-3c8b-11eb-8f3b-7bc3c272cee5.png)


## After
![image](https://user-images.githubusercontent.com/1815379/101993800-7c644980-3c8b-11eb-8df3-d13f93bd9b01.png)

# Details

## General
- Add editorconfig for formatting consistency
- Add eslint + prettier for code standardization
- Add babel for ES6 support
- Add webpack for storybook + jest
- Add jest for testing framework
- Cleanup project per standards installed
- Add husky for commit hooks and lintstaged for eslint precommit checks
- Add PR template for project + Github workflows
- Setup react-helmet for page `<head> manipulation
- Remove scaffold entities and replace with hello world example
- Install styled-components and begin project theme
- Add basic tests to confirm configuration works
- Add storybook for component previews + documentation

## Automated Testing
- Unit Tests
  -  Main page render
  - HelloWorld component render
- Integration tests
  - N/A as there is only static content currently

### Manual Testing
1. Run `yarn dev`
2. Go to `http://localhost:3000/`
3. See Hello World page
